static/js/getReviews.js

```diff
@@ -0,0 +1,121 @@
+let reviewsDiv = document.querySelector(".reviews");
+const btnContainer = document.querySelector(".reviews-btn");
+
+let index = 0;
+let pages = [];
+
+// get reviews function
+export default async function getReviews() {
+  const currentPath = window.location.pathname;
+  const res = await fetch("/api" + currentPath + "/reviews/");
+  const reviews = await res.json();
+  pages = paginate(reviews.attributes.reviews.data);
+  setUpUi();
+
+  console.log("pages", pages);
+}
+
+// display reviews on the dom
+
+function showReviewsOnDom(reviews) {
+  const newReviews = reviews
+    .map((review) => {
+      let list = document.createElement("li");
+
+      let allInfo = document.createElement("div");
+      allInfo.classList.add("all-info");
+      let comment = document.createElement("p");
+      comment.innerText = review.attributes.comment;
+
+      let authorDiv = document.createElement("div");
+      authorDiv.classList.add("author");
+
+      let author = document.createElement("small");
+      author.innerText = review.attributes.author;
+
+      const rating = document.createElement("ul");
+      rating.role = "meter";
+      rating.ariaLabel = "rating";
+      rating.ariaValueMin = 0;
+      rating.ariaValueMax = 5;
+      rating.ariaValueNow = review.attributes.rating;
+      rating.ariaValueText = `${review.attributes.rating} out of 5`;
+      rating.classList.add("rating");
+      for (let i = 1; i <= 5; i++) {
+        const star = document.createElement("li");
+        star.classList.add("rating-star");
+
+        if (i <= review.attributes.rating) {
+          star.classList.add("active");
+        }
+
+        rating.append(star);
+      }
+
+      authorDiv.append(author, rating);
+      allInfo.append(authorDiv, comment);
+      list.append(allInfo);
+      //reviewsDiv.appendChild(list);
+
+      return list.innerHTML;
+    })
+    .join("");
+  console.log("newReviews", newReviews);
+
+  reviewsDiv.innerHTML = newReviews;
+}
+
+// pagination function
+function paginate(reviews) {
+  console.log("paginatereviews", reviews);
+
+  const numOfReviewsPerPage = 5;
+  const numberOfPages = Math.ceil(reviews.length / numOfReviewsPerPage);
+  const newReviews = Array.from({ length: numberOfPages }, (_, index) => {
+    const start = index * numOfReviewsPerPage;
+    return reviews.slice(start, start + numOfReviewsPerPage);
+  });
```

Comment:
A thought on the pagination. If I understand this correctly the pagination is handled on the client? 
An alternative to this is to handle how many/what reviews to show on moviePage on the backend/server. The logic can be run on our server and then only send in this case 5 reviews to client. And maybe use querystrings in the path like ?page=2 to cycle through the pages. What do you think? :)

Reply:
I think it is a matter of preference. Even if you do the logic on the server, you will still have to do some work in the client side to make it work. If you send just 5 reviews to client as you said, then you won't get more reviews to show on the subsequent pages as stated in the task

Reply:
I agree! Although with querystring you can change what page to show from the client. For example everytime you click on nextPage or previousPage button/link it sends a new request to our server  with querystring ?page=pageCounter where pageCounter is the page you want to display. 
This is just an alternative and the current implementation works good! Good job :)

Reply:
That's true. That is an alternative solution. I will think about that next time

Reply:
I asked Richard about what you suggested and he thinks that's the most appropriate solution. I will try to implement that and see how it goes